### PR TITLE
zcash: ZIP-321 memo parsing + populate spendTargets

### PR DESCRIPTION
**Pull request by j0ntz**
### CHANGELOG

Does this branch warrant an entry to the CHANGELOG?

- [x] Yes
- [ ] No

### Dependencies

 none

### Description

Asana: https://app.asana.com/0/1214252040192565/1214281428820963
Blocks gui work: [EdgeApp/edge-react-gui#6000](https://github.com/EdgeApp/edge-react-gui/pull/6000)

Two zcash changes, both unblocking ZNS flows in the GUI:

**1. Honor ZIP-321 `memo` query param in `parseUri`** ([zip-0321 spec](https://zips.z.cash/zip-0321))

`ZcashTools.parseUri` was delegating to the generic `parseUriCommon`, which only extracts `label`/`message`/`category`/`amount` and silently drops the ZIP-321 `memo` query parameter. Scanning a zcashnames.com claim QR — e.g. `zcash:u1k0…?amount=0.015&memo=<base64url of "ZNS:CLAIM:edge:u1fm0cctk8xycu7a5…">` — produced a transaction with an empty shielded memo, breaking ZNS claim flows. Now base64url-decoded and surfaced as `EdgeParsedUri.uniqueIdentifier`, which the GUI already threads through to `EdgeSpendInfo.memos` / `spendTarget.memo`. Validated ≤ 512 decoded bytes per spec.

**2. Populate `EdgeTransaction.spendTargets` for outgoing txs**

`ZcashEngine.processTransaction` was building `EdgeTransaction` without `spendTargets`, even though the native sync event provides `toAddress` for outgoing transactions. Anything in the GUI that reads `transaction.spendTargets?.[0]?.publicAddress` — recipient display, reverse address lookups (e.g. the ZNS reverse lookup on #6000), etc. — was seeing `undefined` for every zcash tx and silently bailing. Now populated with `[{ publicAddress: toAddress, nativeAmount: value, ... }]` whenever `toAddress != null`. Receives (shielded, `toAddress == null`) are untouched.

**Out of scope** (follow-ups if needed):
- `paramindex` / multi-recipient ZIP-321 URIs
- `req-asset` validation (unknown `req-*` should reject per spec)
- Broader refactor to thread the raw URI to the native SDK via `otherParams.zip321Uri` (would cover all ZIP-321 features for free but is a bigger change).

**Tests:** new `test/zcash/zip321.test.ts` exercises the `decodeZip321Memo` helper — ASCII payload, base64url-specific characters (`-` / `_`), 0/1/2-byte padding, binary bytes, empty input. All 451 existing tests still pass. Verified locally with `yarn prepare`, `yarn lint`, `yarn test`.

---

> [!NOTE]
> **Medium Risk**
> Touches Zcash URI parsing and transaction processing, which can affect how memos/recipients are surfaced to the GUI and could cause edge-case decoding or display regressions if memo inputs are malformed.
> 
> **Overview**
> **Zcash payment URI parsing now honors ZIP-321 `memo` query parameters.** `parseUri` decodes base64url (no padding) memos, enforces the configured memo byte limit, and surfaces the decoded text via `EdgeParsedUri.uniqueIdentifier`.
> 
> **Outgoing Zcash transactions now include recipient info in `EdgeTransaction.spendTargets`.** When the SDK provides `toAddress`, the engine populates a single spend target so the GUI can reliably show recipients and perform reverse address lookups.
> 
> Adds unit tests for ZIP-321 memo decoding and updates the changelog accordingly.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f34b4e492d3559da5cebe0d1bdcfe92c9c3dab24. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/zcash/ZcashTools.ts [hunk 1]
```diff
@@ -27,6 +27,21 @@ import {
...
+ * Per https://zips.z.cash/zip-0321: memo uses base64url alphabet, no padding,
+ * decoded ≤ 512 bytes.
+ */
+export function decodeZip321Memo(b64url: string): Buffer {
+  const standard = b64url.replace(/-/g, '+').replace(/_/g, '/')
+  const remainder = standard.length % 4
+  const padded =
+    remainder === 0 ? standard : standard + '='.repeat(4 - remainder)
+  return Buffer.from(padded, 'base64')
+}
```
[the remark this answers is not in the document]
Comment (j0ntz) [reply src/zcash/ZcashTools.ts:38]: Fixed in d51ede9d — replaced the manual base64url with `base64url.parse(b64url, { loose: true })` from the existing `rfc4648` dep, matching the convention used in ZcashEngine, StellarEngine, SuiTools, etc. `loose: true` permits the missing `=` padding ZIP-321 mandates.

File: src/zcash/ZcashTools.ts [hunk 2]
```diff
@@ -174,6 +187,19 @@ export class ZcashTools implements EdgeCurrencyTools {
...
     }
 
+    // ZIP-321 `memo` query parameter — base64url without padding, ≤ 512 bytes
+    // decoded. https://zips.z.cash/zip-0321
+    // Surface as `uniqueIdentifier`, which the GUI threads through to
+    // EdgeSpendInfo.memos (and the legacy `spendTarget.memo`).
+    const memoB64Url = parsedUri.query.memo
+    if (memoB64Url != null && memoB64Url !== '') {
+      const memoBytes = decodeZip321Memo(memoB64Url)
+      if (memoBytes.length > ZIP321_MEMO_MAX_BYTES) {
```
Comment (peachbits) [inline src/zcash/ZcashTools.ts]: we already store max bytes in the memo options in the info

**Review by peachbits — APPROVED** [review]
approved, please fix the length const lookup to use memo options

File: src/zcash/ZcashTools.ts (hunk 2 above)
Comment (j0ntz) [reply src/zcash/ZcashTools.ts]: Switched to `this.currencyInfo.memoOptions?.[0]?.maxLength` and removed the local `ZIP321_MEMO_MAX_BYTES` constant — the limit is now sourced from `zcashInfo.memoOptions`.

**Comment by j0ntz** [comment]
Length lookup now reads from `memoOptions[0].maxLength` instead of the hardcoded constant.
